pkg/controllers/user/nsserviceaccount/nssvcaccnt.go

```diff
@@ -0,0 +1,87 @@
+package nsserviceaccount
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type namespaceSvcAccountHandler struct {
+	clusterCtx *config.UserContext
+}
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	logrus.Debugf("Registering namespaceHandler for checking default serviceaccount")
+	nsh := &namespaceSvcAccountHandler{
+		clusterCtx: cluster,
+	}
+	cluster.Core.Namespaces("").AddHandler(ctx, "namespaceHandler", nsh.Sync)
+}
+
+func (nsh *namespaceSvcAccountHandler) Sync(key string, ns *corev1.Namespace) (runtime.Object, error) {
+	if ns == nil {
+		return nil, nil
+	}
+	logrus.Debugf("namespaceSvcAccountHandler: Sync: key=%v, ns=%+v", key, *ns)
+
+	//handle default svcAccount of system namespaces
+	err := nsh.handleSystemNS(key)
+	if err != nil {
+		logrus.Errorf("namespaceSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=%v, err=%v", key, err)
+		return nil, nil
```

Comment:
nit: Should this be returning an error? If not, we can just omit this line as it already returns `nil, nil`

Reply:
Yeah will remove this; we need not return error here since its not something we should try over and over aggressively - the periodic controller sync should call it again.